.github/workflows/nightly.yml

```diff
@@ -12,6 +12,110 @@ permissions:
   contents: read
 
 jobs:
+  build-clean-caches-clippy:
+    name: Build clean caches
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+    steps:
+      - uses: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5 # v4
+
+      - uses: ./.github/actions/install-rust-toolchain
+        with:
+          components: clippy
+
+      - uses: ./.github/actions/rust-cache
+        with:
+          identifier: 'clippy'
+          restore-strategy: 'exact'
+          save-cache: true
+      
+      - name: cargo clippy with compiler warnings
+        run: cargo clippy --workspace --all-targets -- -D warnings -D clippy::all
+
+  build-clean-caches-test-dev:
+    name: Build clean caches
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+    steps:
+      - uses: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5 # v4
+
+      - uses: ./.github/actions/rust-cache
+        with:
+          identifier: 'test-dev'
+          restore-strategy: 'exact'
+          save-cache: true
+
+      - uses: ./.github/actions/install-rust-toolchain
+      
+      - name: cargo clippy with compiler warnings
+        run: |
+          cargo test --no-run --workspace --exclude 'newrelic_agent_control' --all-targets -- --include-ignored
+          make -C agent-control test/cache
```

Comment:
Making sure that we have a cache with every artifact for test is hard. We need the artifact for every possible combination, so to speak, that we want to test.

Instead of trying to be smart, I ended up calling here every `cargo test` we are calling in other places with `--no-run` to just compile them.